src/FocusApp.Client/Views/Social/SocialPage.cs

```diff
@@ -119,12 +159,45 @@ protected override async void OnAppearing()
             loginPopup.OriginPage = nameof(SocialPage);
         }
 
+        // Retrieve Friends from API
+        List<FriendListModel> friendsList;
+
+        var query = new GetAllFriendsQuery
+        {
+            UserId = _authenticationService.CurrentUser.Id
+        };
+        friendsList = await _client.GetAllFriends(query, default);
+
+        _friendsListView.ItemsSource = friendsList;
+
         base.OnAppearing();
     }
 
+    // We call this function from friends popup to refresh the friends list
+    public async void RepopulateFriendsList()
+    {
+        // Retrieve Friends from API
+        List<FriendListModel> friendsList;
+
+        var query = new GetAllFriendsQuery
+        {
+            UserId = _authenticationService.CurrentUser.Id
+        };
+        friendsList = await _client.GetAllFriends(query, default);
```

Comment:
Since this doesn't return an ApiResponse, could we get a try-catch with logging around it?